src/environment.ts

```diff
@@ -145,117 +105,53 @@ export class SecureEnvironment implements MembraneBroker {
         WeakMapSet(this.blueMap, blue, red);
     }
 
-    remap(redValue: RedValue, blueValue: BlueValue, blueDescriptors?: PropertyDescriptorMap) {
-        this.setRefMapEntries(redValue, blueValue);
-        if (!isUndefined(blueDescriptors)) {
-            const keys = ownKeys(blueDescriptors);
-            for (let i = 0, len = keys.length; i < len; i += 1) {
-                const key = keys[i];
-                // TODO: this whole loop needs cleanup and simplification avoid
-                // overriding ECMAScript global keys.
-                if (SetHas(ESGlobalKeys, key) || !hasOwnProperty(blueDescriptors, key)) {
-                    continue;
-                }
-
-                // avoid poisoning by only installing own properties from blueDescriptors
-                // @ts-ignore PropertyDescriptorMap def defines properties as being only of string type
-                const blueDescriptor = assign(ObjectCreate(null), blueDescriptors[key]);
-                if ('value' in blueDescriptor) {
-                    // TODO: maybe we should make everything a getter/setter that way
-                    // we don't pay the cost of creating the proxy in the first place
-                    blueDescriptor.value = this.getRedValue(blueDescriptor.value);
-                } else {
-                    // Use the original getter to return a red object, but if the 
-                    // sandbox attempts to set it to a new value, this mutation will
-                    // only affect the sandbox's global object, and the getter will
-                    // start returning the new provided value rather than calling onto
-                    // the blue realm. This is to preserve the object graph of the
-                    // blue realm.
-                    const env = this;
-                    const { get: originalGetter } = blueDescriptor;
-
-                    let currentGetter = () => undefined;
-                    if (isFunction(originalGetter)) {
-                        const originalOrDistortedGetter: () => any = WeakMapGet(this.distortionMap, originalGetter) || originalGetter;
-                        currentGetter = function(this: any): RedValue {
-                            const value: BlueValue = apply(originalOrDistortedGetter, env.getBlueValue(this), emptyArray);
-                            return env.getRedValue(value);
-                        };
-                    }
-
-                    blueDescriptor.get = function(): RedValue {
-                        return apply(currentGetter, this, emptyArray);
+    remap(redValue: RedValue, blueValue: BlueValue, blueDescriptors: PropertyDescriptorMap) {
+        const broker = this;
+        for (const key in blueDescriptors) {
```

Comment:
`for-in` susceptible to prototype pollution?

Reply:
will fix.